ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -137,10 +163,104 @@ public static Bytes32 getAttestationMessageToSign(AttestationData attestationDat
   }
 
   public static int getDomain(BeaconState state, AttestationData attestationData) {
-    return BeaconStateUtil.get_domain(
+    return get_domain(
             state.getFork(),
             BeaconStateUtil.slot_to_epoch(attestationData.getSlot()),
             Constants.DOMAIN_ATTESTATION)
         .intValue();
   }
+
+  public static boolean is_slashable_attestation_data(AttestationData data_1, AttestationData data_2){
+    //Check if ``data_1`` and ``data_2`` are slashable according to Casper FFG rules.
+    return (
+      //case 1: double vote || case 2: surround vote
+      (!data_1.equals(data_2) && data_1.getTarget_epoch().equals(data_2.getTarget_epoch()) ||
+      (data_1.getSource_epoch().compareTo(data_2.getSource_epoch()) < 0 && data_2.getTarget_epoch().compareTo(data_1.getTarget_epoch()) < 0))
+    );
+  }
+
+  public static IndexedAttestation convert_to_indexed(BeaconState state, Attestation attestation){
+    //Convert ``attestation`` to (almost) indexed-verifiable form.
+    List<UnsignedLong> attestations = get_attesting_indices(state, attestation.getData(), attestation.getAggregation_bitfield());
```

Comment:
Would you mind changing the return value of `get_attesting_indices` to a List of Integers?

Reply:
Done